src/dreamchecker/lib.rs

```diff
@@ -1043,17 +1043,37 @@ impl<'o, 's> AnalyzeProc<'o, 's> {
         }
     }
 
-    fn visit_unary(&mut self, rhs: Analysis<'o>, op: &UnaryOp) -> Analysis<'o> {
+    // crement = increment/decrement
+    fn check_crement(&mut self, rhs: Analysis<'o>, location: Location, crement: &str) -> Analysis<'o> {
+        let typeerror;
+        match rhs.static_ty {
+            StaticType::None => {
+                return Analysis::empty()
+            },
+            StaticType::Type(typeref) => {
+                if let Some(proc) = typeref.get_proc(crement) {
+                    return Analysis::empty()
+                }
+                typeerror = typeref.get().pretty_path();
+            },
+            StaticType::List { list, .. } => {
+                typeerror = "list";
+            },
+        };
+        error(location, format!("attempted {} on a {}", crement, typeerror))
+            .register(self.context);
+        return Analysis::empty()
+    }
+
+    fn visit_unary(&mut self, rhs: Analysis<'o>, op: &UnaryOp, location: Location) -> Analysis<'o> {
         match op {
             // !x just evaluates the "truthiness" of x and negates it, returning 1 or 0
             UnaryOp::Not => Analysis::from(assumption_set![Assumption::IsNum(true)]),
+            UnaryOp::PreIncr | UnaryOp::PostIncr => { return self.check_crement(rhs, location, "operator++"); },
```

Comment:
You can write `=> self.check_crement(...)` rather than having to specifically put `return` here.